pkg/options/gotmpl/go_template_applier.go

```diff
@@ -32,6 +33,8 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
+var multiDoc = regexp.MustCompile("---(\n|$)")
```

Comment:
Match the beginning of the line as well.

Reply:
`^` will cause the regex match just the beginning of entire string. 

https://play.golang.org/p/NT2W17pdTVt

Reply:
`(\n|^)---(\n|$)` or don't use regex? Is `foo: a---` valid yaml?

Reply:
Ah, right. Done.